lib/vehicle_factory.rb

```diff
@@ -0,0 +1,27 @@
+class VehicleFactory
+  attr_reader :created_vehicles
+
+  def initialize
+    @created_vehicles = created_vehicles
+  end
+  
+  def create_vehicles(registrations)
+    created_vehicles = []
+
+    registrations.map do |registration|
+      specific_vehicle = {
+        engine: :ev, 
+        make: registration[:make],
+        model: registration[:model],
+        registration_date: nil,
+        vin: registration[:vin_1_10], 
+        year: registration[:model_year]
+      }
+
+    vehicle = Vehicle.new(specific_vehicle)
+    created_vehicles << vehicle
+    end
+    created_vehicles
+  end
```

Comment:
Nice job with this, however it's not wired up quite right. On line 5, you're setting your instance variable equal to the local variable of `created_vehicles` that that variable has not been assigned so it's currently nil. And then in your create_vehicles method, you're not reseting the value of your instance variable so it's not being updated. This method will still work by itself though since the return value of the method is the array of vehicle objects. Also, you don't need the accumulator on line 9 since you're using .map which already returns the transformed array. 
Good work with setting up that hash and creating the new Vehicle objects.